internal/vehicle/ford.go

```diff
@@ -134,36 +169,135 @@ func (v *Ford) vehicles() ([]string, error) {
 	return vehicles, err
 }
 
-// chargeState implements the api.Vehicle interface
-func (v *Ford) chargeState() (float64, error) {
-	var resp struct {
-		VehicleStatus struct {
-			Battery struct {
-				BatteryStatusActual struct {
-					Value int
-				}
-			}
-			BatteryFillLevel struct {
-				Value float64
+// calculateAge parses Ford API timestamp and returns age
+func (v *Ford) calculateAge(timestamp string) (age time.Duration, err error) {
+	var timestampTime time.Time
+	const dateFormat = "01-02-2006 15:04:05" // time format used by Ford API, time is in UTC
+	timestampTime, err = time.Parse(dateFormat, timestamp)
+	age = time.Since(timestampTime)
+
+	return age, err
+}
+
+// vehicleStatus performs a /status response to the Ford API and triggers a refresh if
+// the received status is too old
+func (v *Ford) vehicleStatus() (res fordVehicleStatus, err error) {
+	uri := fmt.Sprintf("%s/api/vehicles/v3/%s/status", fordAPI, v.vin)
+
+	var req *http.Request
+	req, err = v.request(http.MethodGet, uri)
+	if err == nil {
+		err = v.DoJSON(req, &res)
+	}
+
+	var statusAge time.Duration
+	if err == nil {
+		statusAge, err = v.calculateAge(res.VehicleStatus.LastRefresh)
+	}
+
+	if err == nil && statusAge > fordOutdatedAfterMins*time.Minute {
+		// received data is considered outdated, server is requested to poll updated status from vehicle
+		v.log.DEBUG.Printf("Vehicle Status is outdated (age %v > %dm), requesting refresh", statusAge, fordOutdatedAfterMins)
+		var refreshRes fordVehicleStatus
+		var refreshErr error
+		refreshRes, refreshErr = v.vehicleStatusRefresh()
+		if refreshErr == nil {
+			res = refreshRes
+			statusAge, err = v.calculateAge(res.VehicleStatus.LastRefresh)
+			v.log.DEBUG.Printf("Refreshed Status Age: %v", statusAge)
+		} else {
+			// error occured while trying to get updated data; log and continue with cached data
+			v.log.WARN.Print(refreshErr)
```

Comment:
warum wird der Fehler nicht zurück gegeben?

Reply:
siehe Kommentar oben - ich habe in Schritt 1 einen Status von der API bekommen; der war mir zu alt, daher ein Update versucht. Update fehlgeschlagen -> ich habe immer noch den "leicht veralteten" Status -> besser als nichts; falls Schritt 1 schon fehlgeschlagen ist und gar nichts von der API kam bricht die Funktion mit einem Fehler ab

Reply:
habe nochmal nachgedacht und das vereinfacht - wenn der Refresh fehlschlägt (z.B. Fahrzeug im Funkloch) haben wir eben keinen Status, es hilft dann auch nicht irgendeinen alten Wert zu haben, der sich nicht verändert; macht den Code viel viel kürzer